primitives:
  trigger:
    type: base
    input_ports: 
      trigger: bool
    output_ports: 
      status: bool
  move2pose:
    type: trajectory
    input_ports: 
      pose: Pose
    output_ports: 
      status: bool
  Move:
      type: trajectory
      input_ports: 
        pose: Pose
      output_ports: 
        status: bool
  Pick:
    type: composite
    input_ports:
      pose: Pose
      trigger: bool
    output_ports:
      status: bool
  Pick_Front:
    type: composite
    input_ports:
      pose: Pose
      trigger: bool
    output_ports:
      status: bool
  Place:
    type: composite
    input_ports:
      pose: Pose
      trigger: bool
    output_ports:
      status: bool